src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -218,6 +329,12 @@ private static ConsoleColor ColorAttributeToConsoleColor(Interop.mincore.Color c
             return (ConsoleColor)c;
         }
 
+        private static Interop.mincore.CONSOLE_SCREEN_BUFFER_INFO GetBufferInfo()
+        {
+            bool unused;
+            return GetBufferInfo(true, out unused);
+        }
+
```

Comment:
I have manually tested the scenarios. However, since most of the scenarios don't work in case of redirection, the value add is minimum.